source/configuration/modules/imfile.rst

```diff
@@ -1,570 +1,408 @@
-   it is strongly suggested *not* to modify monitored files with
-   editors. In any case, in such a situation, it makes sense to
-   disable state file deletion. That also applies to similar use
-   cases.
+================  ==================
+mode:             "inotify|polling"
+Default:          "inotify"
+Legacy:           $InputFileReadMode
+Available since:  8.1.5
+================  ==================
 
-   In general, this parameter should only by set if the users
-   knows exactly why this is required.
+This specifies if imfile is shall run in *inotify* or *polling* mode.
 
-.. index:: 
-   single: imfile;  Ruleset
-.. function:: Ruleset <ruleset> 
+inotify:
 
-   Binds the listener to a specific :doc:`ruleset <../../concepts/multi_ruleset>`.
+**TODO add a description about how inotify works internally**
 
-.. function:: addMetadata [on/off]
+Although lines are read *sooner* than using *polling* mode, there may be a slight delay due to internal processing.
 
-   **Default: see intro section on Metadata**
+polling: it needs more resources and its slower than *inotify*, reason why **mode="inotify" is recommended**.
+Basically there's a timeout at which file is checked for changes.
 
-   This is used to turn on or off the addition of metadata to the
-   message object.
+================  ======================
+pollinginterval:  "seconds"
+Default:          "10"
+Legacy:           $InputFilePollInterval
+Available since:  8.1.5
+================  ======================
 
-.. function:: stateFile [name-of-state-file]
+This setting specifies how often files are checked for changes when using **mode="polling"**. Basically, this mean files are read and rsyslog sleeps for X seconds before being read again.
 
-   **Default: unset**
+Although it's possible, its stongly recommend not to set it to 0 seconds, because in such scenario rsyslog will take far too many resources.
 
-   **This paramater is deprecated.** It still is accepted, but should
-   no longer be used for newly created configurations.
+Input parameters
+****************
 
-   This is the name of this file's state file. This parameter should
-   usually **not** be used. Check the section on "State Files" above
-   for more details.
-
-.. index::
-   single: imfile; reopenOnTruncate
-.. function:: reopenOnTruncate [on/off] (requires v8.16.0+)
-
-   **Default: off**
-
-   This is an **experimental** feature that tells rsyslog to reopen input file
-   when it was truncated (inode unchanged but file size on disk is less than
-   current offset in memory).
-
-.. index::
-   single: imfile; trimLineOverBytes
-.. function:: trimLineOverBytes [number] (requires v8.17.0+)
+============  ===============
+**REQUIRED**
+file:         "/path/to/file"
+Default:      none
+Legacy:       $InputFileName
+============  ===============
 
-   **Default: 0**
+File(s) being processed. This must be an absolute file path (no macros or templates). 
```

Comment:
FYI, I think it can be a relative path to where rsyslog is started, but it's a _really_ bad idea to depend on that.

so I'd leave the comment as-is in the documentation, but it's good for you to know this.